### ISSUE TITLE
Applications remained in the Dash after being closed

### ISSUE THREAD

**Issue by rthery** [pop-os/shell#348]
**Distribution (run `cat /etc/os-release`):**
```
 NAME="Pop!_OS"
VERSION="20.04 LTS"
ID=pop
ID_LIKE="ubuntu debian"
PRETTY_NAME="Pop!_OS 20.04 LTS"
VERSION_ID="20.04"
HOME_URL="https://system76.com/pop"
SUPPORT_URL="http://support.system76.com"
BUG_REPORT_URL="https://github.com/pop-os/pop/issues"
PRIVACY_POLICY_URL="https://system76.com/privacy"
VERSION_CODENAME=focal
UBUNTU_CODENAME=focal
LOGO=distributor-logo-pop-os
```

 
**Related Application and/or Package Version (run `apt policy pop-shell` or provide the latest commit if building locally):**
```
pop-shell:
  Installed: 0.1.0~1589481653~20.04~87a5f44
  Candidate: 0.1.0~1589481653~20.04~87a5f44
  Version table:
 *** 0.1.0~1589481653~20.04~87a5f44 1001
       1001 http://ppa.launchpad.net/system76/pop/ubuntu focal/main amd64 Packages
       1001 http://ppa.launchpad.net/system76/pop/ubuntu focal/main i386 Packages
        100 /var/lib/dpkg/status
```

**Issue/Bug Description:**
Closing a GUI application will leave its icon in the Dash. Left clicking it does nothing, right clicking will freeze the shell.

**Steps to reproduce (if you know):**
1. Open an application, such as the new Extensions in Utilities
2. Close it
3. In the Dash, notice that Extensions is still there
4. Right click to force quit
5. Your shell is now frozen, Ctrl + Alt + F3 is your only solution, or a reboot

**Expected behavior:**
Application is properly removed from the Dash when closed

**Other Notes:**
This is a regression introduced with the last update. I made sure to disable all my other gnome extensions, and I could still reproduce it with only the default system extensions.

**Comment by rthery** [comment]
I think it froze when the application doesn't support multiple window, I have the same issue with Firefox, but I can access the right click menu. I can gather more logs later today if needed.

**Comment by lungtendo** [comment]
I also have noticed this issue and it got worse with the latest update to Pop shell. Initially, it used to happen with VirtualBox icon but not it seems to happen with any app. 

For me this happens only when tiling is turned on.

**Comment by mmstick** [comment]
@lungtendo There's a serious regression in the last update, but I have a fix queued in #344 

Also, this is a duplicate of #220
